.github/workflows/ci-go.yml

```diff
@@ -50,3 +50,33 @@ jobs:
           go get .
           gosec -exclude=G204,G304,G107 -exclude-dir=protobuf/ ./...  
         working-directory: src
+
+  go-ginkgo-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: v1.18
+        
+      - name: Setup Enviroment
+        id: vars
+        run: |
+          if [ ${{ github.ref }} == "refs/heads/dev" ]; then
+            echo ::set-output name=tag::latest
+          else 
+            echo ::set-output name=tag::${GITHUB_REF#refs/*/}
+          fi
+        
+          # setup kubernetes environment
+          ./k3s/install_k3s.sh
+
+      - name: Install Ginkgo
+        run: |
+          go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo
+
+      - name: Run ginkgo tests
```

Comment:
working directory should be `tests` here, it's failing because no ginkgo test suit present the current working directory.